apps/webservice/src/app/[workspaceSlug]/(targets)/targets/TargetPageContent.tsx

```diff
@@ -27,54 +32,62 @@ export const TargetPageContent: React.FC<{
+    if (isEmptyCondition(cond)) setFilter(undefined);
+    if (!isEmptyCondition(cond)) setFilter(cond);
+  };
 
   const { targetId, setTargetId } = useTargetDrawer();
 
   if (targetsAll.isSuccess && targetsAll.data.total === 0)
     return <TargetGettingStarted />;
   return (
     <div className="h-full text-sm">
-      <div className="flex items-center justify-between border-b border-neutral-800 p-1 px-2">
-        <TargetConditionDialog condition={filter} onChange={setFilter}>
-          <div className="flex items-center gap-2">
-            {view == null && (
-              <Button
-                variant="ghost"
-                size="icon"
-                className="flex h-7 w-7 flex-shrink-0 items-center gap-1 text-xs"
-              >
-                <IconFilter className="h-4 w-4" />
-              </Button>
-            )}
+      <div className="flex h-[41px] items-center justify-between border-b border-neutral-800 p-1 px-2">
+        <div className="flex items-center gap-2">
+          <TargetConditionDialog condition={filter} onChange={onFilterChange}>
+            <div className="flex items-center gap-2">
+              {view == null && (
+                <Button
+                  variant="ghost"
+                  size="icon"
+                  className="flex h-7 w-7 flex-shrink-0 items-center gap-1 text-xs"
+                >
+                  <IconFilter className="h-4 w-4" />
+                </Button>
+              )}
 
-            {filter != null && view == null && (
-              <TargetConditionBadge condition={filter} />
-            )}
-            {view != null && (
-              <>
-                <span>{view.name}</span>
-                <TargetViewActionsDropdown view={view}>
-                  <Button
-                    variant="ghost"
-                    size="icon"
-                    className="flex h-7 w-7 flex-shrink-0 items-center gap-1 text-xs"
-                  >
-                    <IconDots className="h-4 w-4" />
-                  </Button>
-                </TargetViewActionsDropdown>
-              </>
-            )}
-          </div>
-        </TargetConditionDialog>
+              {filter != null && view == null && (
+                <TargetConditionBadge condition={filter} />
+              )}
+              {view != null && (
+                <>
+                  <span>{view.name}</span>
+                  <TargetViewActionsDropdown view={view}>
+                    <Button
+                      variant="ghost"
+                      size="icon"
+                      className="flex h-7 w-7 flex-shrink-0 items-center gap-1 text-xs"
+                    >
+                      <IconDots className="h-4 w-4" />
+                    </Button>
+                  </TargetViewActionsDropdown>
+                </>
+              )}
+            </div>
+          </TargetConditionDialog>
+          {!targets.isLoading && targets.isFetching && (
```

Comment:
only show the skeletons on the first load. then, show the animated laoder next to the new filter, and keep the previous data